src/config.ts

```diff
@@ -188,6 +188,7 @@ export type RetryConfig = {
   retryDelayMs: number; // Retry delay = 2^retryCount * retryDelayMs (exponential) or retryCount * retryDelayMs (linear)
   maxRetryDelayMs: number; // Maximum delay between retries
   backoff?: 'exponential' | 'linear'; // used to determine retry backoff duration (see retryDelayMs)
+  shouldRetry?: () => boolean;
```

Comment:
This callback should take all the parameters plus the result of the internal `shouldRetry` helper. This way, the custom callback can evaluate whether or not is should override the internal result:

```suggestion
  shouldRetry?: (
    retryConfig: RetryConfig | null | undefined,
    retryCount: number,
    isTimeout: boolean,
    httpStatus: number | undefined,
    retry: boolean
  ) => boolean;
```